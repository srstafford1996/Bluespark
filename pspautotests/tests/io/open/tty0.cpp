#include <common.h>
#include <pspstdio.h>
#include <pspiofilemgr.h>

extern "C" int main(int argc, char *argv[]) {
	// Let's be fair... need to simulate the stderr/stdout pipes.
	if (RUNNING_ON_EMULATOR) {
		sceIoOpen("fds.prx", PSP_O_RDONLY, 0);
	}

	checkpointNext("sceIoOpen tty0:");

	checkpoint("  Read only: %08x", sceIoOpen("tty0:", PSP_O_RDONLY, 0));
	checkpoint("  Write only: %08x", sceIoOpen("tty0:", PSP_O_WRONLY, 0));
	checkpoint("  Read write: %08x", sceIoOpen("tty0:", PSP_O_RDWR, 0));
	checkpoint("  Create: %08x", sceIoOpen("tty0:", PSP_O_WRONLY | PSP_O_CREAT, 0));
	checkpoint("  Truncate: %08x", sceIoOpen("tty0:", PSP_O_WRONLY | PSP_O_TRUNC, 0));
	checkpoint("  Append: %08x", sceIoOpen("tty0:", PSP_O_WRONLY | PSP_O_APPEND, 0));
	checkpoint("  With name: %08x", sceIoOpen("tty0:foobar.txt", PSP_O_RDONLY, 0));
	checkpoint("  TTY1: %08x", sceIoOpen("tty1:", PSP_O_RDONLY, 0));

	int fd = sceIoOpen("tty0:", PSP_O_WRONLY, 0);
	sceIoWrite(fd, "test\n", 5);

	return 0;
}