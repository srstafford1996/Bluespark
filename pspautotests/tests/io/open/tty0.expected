[x] sceIoOpen tty0:
[x]   Read only: 00000005
[x]   Write only: 00000006
[x]   Read write: 00000007
[x]   Create: 00000008
[x]   Truncate: 00000009
[x]   Append: 0000000a
[x]   With name: 0000000b
[x]   TTY1: 0000000c
